engine/src/main/java/org/hibernate/validator/internal/engine/validationcontext/ParameterExecutableValidationContext.java

```diff
@@ -85,13 +87,16 @@ public Executable getExecutable() {
 		return executableMetaData;
 	}
 
-	private static boolean buildDisableAlreadyValidatedBeanTracking(Optional<ExecutableMetaData> executableMetaData) {
+	private static boolean buildProcessedBeansTrackingEnabled(ProcessedBeansTrackingStrategy processedBeansTrackingStrategy,
+			Executable executable,
+			Optional<ExecutableMetaData> executableMetaData) {
 		if ( !executableMetaData.isPresent() ) {
 			// the method is unconstrained so there's no need to worry about the tracking
 			return false;
 		}
 
-		return !executableMetaData.get().getValidatableParametersMetaData().hasCascadables();
+		return processedBeansTrackingStrategy.isEnabledForReturnValue( executable,
```

Comment:
Should the above line be:

`return processedBeansTrackingStrategy.isEnabledForParameters( executable,`
?

Reply:
You're absolutely right, I copy/pasted things a bit too fast.